Lib/urllib/parse.py

```diff
@@ -436,7 +436,8 @@ def urlsplit(url, scheme='', allow_fragments=True):
             if url[:2] == '//':
                 netloc, url = _splitnetloc(url, 2)
                 if (('[' in netloc and ']' not in netloc) or
-                        (']' in netloc and '[' not in netloc)):
+                        (']' in netloc and '[' not in netloc) or
+                        ('[' in netloc and netloc.index('[') != 0)):
                     raise ValueError("Invalid IPv6 URL")
```

Comment:
_splitnetloc() is called 2 times and the same code to validate the IPv6 address is duplicated, whereas you only fix one place. IMHO it would be better to move the check into _splitnetloc().